MYPY_REPORT.txt

```diff
@@ -0,0 +1,3 @@
+mimesis/providers/units.py:299: error: Argument 1 to "choice" of "Random" has incompatible type "List[str]"; expected "Sequence[Dict[str, List[str]]]"
+mimesis/providers/generic.py:71: error: Item "function" of "Union[Callable[..., Any], Any]" has no attribute "Meta"
```

Comment:
You are using a class, not function.

Reply:
`Provider` should be a protocol: http://mypy.readthedocs.io/en/latest/class_basics.html#simple-user-defined-protocols

Reply:
@sobolevn Should i create `protoclos.py` or just create it in `typing.py`?

Reply:
IMHO stick to `typing.py`

Reply:
@sobolevn Seems like i have not `typing_extensions` module in my version of Python 3.5.2 on Ubuntu 16.04:

![](https://i.imgur.com/GXklYQB.png)

Reply:
https://github.com/python/typing/issues/467

Reply:
So, it's mean that we should add third-party dependencies and i'm not sure that it's worth it. I will just ignore places where we need Protocol.